src/Sylius/Bundle/ApiBundle/Validator/CatalogPromotion/ForProductsScopeValidator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\CatalogPromotion;
+
+use Sylius\Bundle\ApiBundle\SectionResolver\AdminApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Bundle\CoreBundle\Validator\CatalogPromotionScope\ScopeValidatorInterface;
+use Sylius\Bundle\CoreBundle\Validator\Constraints\CatalogPromotionScope;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+use Webmozart\Assert\Assert;
+
+final class ForProductsScopeValidator implements ScopeValidatorInterface
+{
+    public function __construct(
+        private ScopeValidatorInterface $baseScopeValidator,
+        private SectionProviderInterface $sectionProvider
+    ) {
+    }
+
+    public function validate(array $configuration, Constraint $constraint, ExecutionContextInterface $context): void
+    {
+        /** @var CatalogPromotionScope $constraint */
+        Assert::isInstanceOf($constraint, CatalogPromotionScope::class);
+
+        if (!$this->sectionProvider->getSection() instanceof AdminApiSection) {
+            $this->baseScopeValidator->validate($configuration, $constraint, $context);
+
+            return;
+        }
```

Comment:
Thats really minor thing, but `FixedDiscountActionValidator` and `ForTaxonsScopeValidator` have different order of assert and section check

Reply:
Caring about minor things is the way to be a decent human being 🖖 Gonna fix it soon